test/e2e/validation_test.go

```diff
@@ -46,6 +46,39 @@ type winService struct {
 	description string
 }
 
+// testKubeletPriority tests if kubelet priority class is set to "AboveNormal"
+func testKubeletPriority(t *testing.T) {
+	tc, err := NewTestContext()
+	require.NoError(t, err)
+	requiredPriorityClass := "AboveNormal"
+
+	if len(gc.allNodes()) == 0 {
+		t.Skip("0 expected nodes")
+	}
+	for _, node := range gc.allNodes() {
+		t.Run("nodes", func(t *testing.T) {
+			out, err := tc.getKubeletPriorityClass(&node)
+			require.NoError(t, err, "error getting kubelet priority class")
+			assert.Containsf(t, out, requiredPriorityClass, "node %s missing required kubelet priority class",
+				node.GetName())
+		})
+	}
+}
+
+// getKubeletPriorityClass returns the PriorityClass of the kubelet service
+func (tc *testContext) getKubeletPriorityClass(node *core.Node) (string, error) {
+	command := "Get-Process kubelet | Select-Object PriorityClass"
+	addr, err := controllers.GetAddress(node.Status.Addresses)
+	if err != nil {
+		return "", fmt.Errorf("error getting node address: %w", err)
+	}
+	out, err := tc.runPowerShellSSHJob("kubelet-query", command, addr)
```

Comment:
Please dont name this the same as other jobs in the tests. That makes figuring out the correct pod log to check hard.

lets be specific here, something like "kubelet-priority-class-query"